python/test/unit/mesh/test_higher_order_mesh.py

```diff
@@ -635,7 +634,7 @@ def test_gmsh_input_2d(order, cell_type):
         gmsh_cell_id = gmsh.model.mesh.getElementType("quadrangle", order)
     gmsh.finalize()
 
-    cells = cells[:, cell_perm_array(cell_type, cells.shape[1])]
+    cells = cells[:, cell_perm_array(cell_type, cells.shape[1])].copy()
```

Comment:
Comment on why this copy is necessary would be good (I guess contiguity?).

Reply:
I don't think we really need a comment on this in the test.

Yes, the issue is contiguous storage.

Reply:
This was causing crashes without a copy. I think permuting the indices is just a "view" which does not work when accessing memory directly.